### PR TITLE
toolchain/cflags: enable dwarf compression (save 50% HDD)

### PR DESCRIPTION
**Pull request by jcarrano**
### Contribution description

Use the `-gz` option to compress ELF sections containing DWARF information.

**This saves around 50% of disk space, without any side effects.**

See https://gcc.gnu.org/onlinedocs/gcc-9.2.0/gcc/Debugging-Options.html#Debugging-Options for more infomation on this option.

Some platforms have an outdated toolchain that does not support `-gz` sothe flag is blacklisted there. Even then, the results are quite impressive.

It would be super-awesome if the AVR, MSP and ESP toolchains could be updated to a version with support for this feature. AFAIK those versions are out there, it is just that we are not using them.

### Testing procedure

I used @cladmi's [`buildtest` branch](https://github.com/cladmi/RIOT/tree/wip/du/buildtest) with this change and compiled the `examples/default` application:

```
$ BUILD_IN_DOCKER=1 DOCKER="sudo docker" make -C examples/default buildtest-indocker
```

The size was obtained with:

```
$ find output -name "*.bin.bindirsize" -type f -exec tail  -n1 '{}' \; | cut -f 1 | awk '{s+=$1} END {printf "%.0f", s}'
```

Results:

- Vanilla: 10328112 KB (**~10GB**).
- with -gz: 4982788 KB (**~5GB**).

### Issues/PRs references

This was inspired by #8496.
See #10195 for another 50% savings (TODO: combine the two and see if we get 75% savings)

### CODE REVIEW

**Comment by cladmi** [comment]
I find similar sizes when comparing this PR merged with https://github.com/cladmi/RIOT/tree/wip/du/buildtest and this pr master merged with https://github.com/cladmi/RIOT/tree/wip/du/buildtest.

I forgot to add `RIOT_CI_BUILD=1` so have tiny bytes difference.

I will do a `llvm` comparison too.

**Comment by cladmi** [comment]
With `llvm`, I get some compilation errors, both in master and this PR but are unrelated with this change.

So `llvm` is correctly handling the `-gz` for all the tested architectures.

It somehow looks like `llvm` does some compression by default as the size difference is not as big, but it is still present.

PR: 421296 (4.2GB)
master: 455588 (4.6GB)


For reference the compilation errors:

<details><summary>nucleo-f767zi with cpu/stm32_common/periph/eth.c</summary>

```
Building for b-l475e-iot01a ... /data/riotbuild/riotbase/cpu/stm32_common/periph/i2c_1.c:315:49: error: format specifies type 'unsigned long' but the argument has type                                                                                                                   
      'uint32_t' (aka 'unsigned int') [-Werror,-Wformat]                                                                                                                                                                                                                                  
    DEBUG("[i2c] start: Setting CR2=0x%08lX\n", cr2);                                                                                                                                                                                                                                     
                                      ~~~~~     ^~~                                                                                                                                                                                                                                       
                                      %08X                                                                                                                                                                                                                                                
/data/riotbuild/riotbase/core/include/debug.h:97:50: note: expanded from macro 'DEBUG'                                                                                                                                                                                                    
#define DEBUG(...) if (ENABLE_DEBUG) DEBUG_PRINT(__VA_ARGS__)                                                                                                                                                                                                                             
                                                 ^~~~~~~~~~~                                                                                                                                                                                                                              
/data/riotbuild/riotbase/core/include/debug.h:56:20: note: expanded from macro 'DEBUG_PRINT'                                                                                                                                                                                              
            printf(__VA_ARGS__); \                                                                                                                                                                                                                                                        
                   ^~~~~~~~~~~                                                                                                                                                                                                                                                            
/data/riotbuild/riotbase/cpu/stm32_common/periph/i2c_1.c:380:60: error: format specifies type 'unsigned long' but the argument has type                                                                                                                                                   
      'uint32_t' (aka 'unsigned int') [-Werror,-Wformat]                                                                                                                                                                                                                                  
            DEBUG("[i2c] wait_isr_set: ISR 0x%08lX set\n", mask);                                                                                                                                                                                                                         
                                             ~~~~~         ^~~~                                                                                                                                                                                                                           
                                             %08X                                                                                                                                                                                                                                         
/data/riotbuild/riotbase/core/include/debug.h:97:50: note: expanded from macro 'DEBUG'                                                                                                                                                                                                    
#define DEBUG(...) if (ENABLE_DEBUG) DEBUG_PRINT(__VA_ARGS__)                                                                                                                                                                                                                             
                                                 ^~~~~~~~~~~                                                                                                                                                                                                                              
/data/riotbuild/riotbase/core/include/debug.h:56:20: note: expanded from macro 'DEBUG_PRINT'                                                                                                                                                                                              
            printf(__VA_ARGS__); \                                                                                                                                                                                                                                                        
                   ^~~~~~~~~~~                                                                       
```
</details>

<details><summary>b-l475e-iot01a with cpu/stm32_common/periph/i2c_1.c</summary>

```
Building for nucleo-f767zi ... /data/riotbuild/riotbase/cpu/stm32_common/periph/eth.c:101:31: error: format specifies type 'unsigned long' but the argument has type                                                                                                                      
      'uint32_t' (aka 'unsigned int') [-Werror,-Wformat]                                                                                                                                                                                                                                  
    DEBUG("stm32_eth: %lx\n", ETH->MACMIIDR);                                                                                                                                                                                                                                             
    ~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~                                                                                                                                                                                                                                              
                      %x                                                                                                                                                                                                                                                                  
/data/riotbuild/riotbase/cpu/stm32f7/include/vendor/stm32f767xx.h:1581:29: note: expanded from macro 'ETH'                                                                                                                                                                                
#define ETH                 ((ETH_TypeDef *) ETH_BASE)                                                                                                                                                                                                                                    
                            ^                                                                                                                                                                                                                                                             
/data/riotbuild/riotbase/core/include/debug.h:97:50: note: expanded from macro 'DEBUG'                                                                                                                                                                                                    
#define DEBUG(...) if (ENABLE_DEBUG) DEBUG_PRINT(__VA_ARGS__)                                                                                                                                                                                                                             
                                     ~~~~~~~~~~~~^~~~~~~~~~~~                                                                                                                                                                                                                             
/data/riotbuild/riotbase/core/include/debug.h:56:20: note: expanded from macro 'DEBUG_PRINT'                                                                                                                                                                                              
            printf(__VA_ARGS__); \                                                                                                                                                                                                                                                        
                   ^~~~~~~~~~~                                                                                                                                                                                                                                                            
1 error generated.
```
</details>

**Comment by cladmi** [comment]
Let's wait for CI to find if it has issues with toolchain and the flag in some cases.

This should have no impact on the murdock build as it disables compiling debug https://github.com/RIOT-OS/RIOT/pull/12097

**Comment by cladmi** [comment]
CI is ok but failing due to unrelated bug. Fix to disable unreliable test: https://github.com/RIOT-OS/RIOT/pull/12154

**Comment by cladmi** [comment]
I tested `examples/hello-world` too and got from `10328104` to `3441808` so 10GB to 3.4GB, so even better.
And with `hello-world` all boards are compiled there (which is not the case for `default`).

**Comment by benpicco** [comment]
CI is happy and the change is minimal.
Is there anything speaking against this?

**Comment by kaspar030** [comment]
> CI is happy and the change is minimal.
> Is there anything speaking against this?

not from my side. CI shouldn't create debug info, anyways.

**Comment by jcarrano** [comment]
@benpicco if we are extremely paranoid, I can do a build on @cladmi's paper-CI which uses an unmodified setup (mmm, is it a Bio-CI then?) but I cannot use all board because the resources are rather limited and I would not be done ever. In any case, Murdock is not compiling with debug info, but it is _not_ filtering out the `-gz` flag meaning all toolchains are getting it (unless blacklisted). The fact that they still run means the `-gz` flag is recognized even if not used.

**Comment by benpicco** [comment]
The same `arm-none-eabi-gcc` toolchain is used for all boards that are not blacklisted here, so I think there is little point in testing them individually.

**Comment by jcarrano** [comment]
@benpicco thanks for the review, paper-ci's HDDs will be a lot happier with this change.

**Comment by kaspar030** [comment]
This somehow makes ccache almost ineffective on CI. Maybe you noticed almost doubled build times today? My [first thought](https://github.com/RIOT-OS/RIOT/pull/10661#issuecomment-529918452) turned out to be wrong. During the day, I cleared the cache on one of the workers, but build times stayed double. So cache thrashing was no issue, (and should't be actually as even the oldest PRs get rebased to master and are thus not much different).

Then I ran a script that ran a limited build (samr21-xpro only) on one worker, once with clean cache, then one more time with hot cache, on all merged commits since the last nightly.

Long story short, 3aa8bc0d7001f2304225295fe53e46ba58e3d22d (this PR's merge commit) somehow makes ccache ineffective.

@jcarrano can we maybe just add "-gz" to CFLAGS_DBG?
